### DISCUSSION TITLE
Python 3.10 AttributeError: 'dict' object has no attribute 'header_serializer'

### DISCUSSION THREAD

**Discussion by raulroscode** [aws-powertools/powertools-lambda-python#2022] [General]
Hello, I have updated my runtime python version to 3.10 and I'm getting this error: 

`AttributeError: 'dict' object has no attribute 'header_serializer'. `

```

cors = <aws_lambda_powertools.event_handler.api_gateway.CORSConfig object at 0x127ec6800>

    def build(self, event: BaseProxyEvent, cors: Optional[CORSConfig] = None) -> Dict[str, Any]:
        """Build the full response dict to be returned by the lambda"""
        self._route(event, cors)
    
        if isinstance(self.response.body, bytes):
            logger.debug("Encoding bytes response with base64")
            self.response.base64_encoded = True
            self.response.body = base64.b64encode(self.response.body).decode()
    
        return {
            "statusCode": self.response.status_code,
            "body": self.response.body,
            "isBase64Encoded": self.response.base64_encoded,
          **event.header_serializer().serialize(headers=self.response.headers, cookies=self.response.cookies),
        }
      AttributeError: 'dict' object has no attribute 'header_serializer'
venv/lib/python3.10/site-packages/aws_lambda_powertools/event_handler/api_gateway.py:252: AttributeError

```
I think that aws-lambda-powertools only supports until python 3.9. Should it work with python 3.10?. Do you know when it will be available for python 3.10 and/or 3.11?

Thanks.

**Comment by heitorlessa** [comment]
hey @raulroscode, thanks a lot for creating this discussion. I'll transform as an issue and mark as a feature request so we can track it accordingly :) 

We haven't tested with 3.10 as it's not yet officially supported as a managed runtime. That said, the error seems slightly misleading as the `event` isn't a `dict`, so I wonder if it's failing due to the `dict` vs `Dict`.

One thing that would be super helpful is to have a GitHub sample to reproduce it quickly. I'd also appreciate any contribution for any backwards compatible fix to support 3.10 in areas we don't - I haven't fully gone through line by line in the [3.10 changes yet](https://docs.python.org/3/whatsnew/3.10.html).

@peterschutt IIRC you've been using 3.10 for quite some time using OCI - have you bumped into this?

**Reply by heitorlessa** [reply]
Issue: https://github.com/awslabs/aws-lambda-powertools-python/issues/2028
